local dependencies

gbisschoff opened the pull request:
Hi, 

So as we briefly discussed in July, I made the dependencies local. This helps to not download the dependencies when connecting to spark. 

Also very useful if behind a company proxy

Commit:
add unit test

Commit:
make dependencies local
update to version 2.0.0

Commit:
Merge branch 'master' of https://github.com/bnosac/spark.sas7bdat

# Conflicts:
#	R/read_sas.R

jwijffels commented:
thanks, can you also update the license as we are now shipping jar files which are under an apache license

jwijffels commented:
cran policy also requires you to provide source code or source code links if shipping to cran: https://cran.r-project.org/web/packages/policies.html + to mention all authors if all source code - that's also something that needs to be done if we want to push again on cran

Commit:
- add Apache 2.0 License for external Java dependencies
- Add section in README that have links to the external Java gGitHub repos
- update spark_read_sas documentation
- update DESCRIPTION file to point to LICENSE file

gbisschoff commented:
Would adding a section called 'External Java Dependencies' that points to the GitHub repos  and adding the Apache License in the LICENSE file  be enough?

jwijffels commented:
This might be good enough for github but cran will reject this. 
As we are now shipping someone else his jar/java files which have their own licenses/copyrights, you need to 

- Indicate the sources where you extracted the jar files from
- You need to extract all copyright information of the java files of these source and add them to the new repository
- An example is here https://github.com/bnosac/crfsuite where this R package ships other people's C++ code with a license compatible with the license of the package
- You can not combine GPL-3 and the Apache license in the DESCRIPTION file. I don't mind changing the license of the R package to apache if needed to have a common license, but can you check both source codes of the jar files for all license/copyright related things in order to add them to the repo

Commit:
add more detail to DESCRIPTION file on java dependency authors
change license from GPL-3 to Apache 2.0 (LICENSE file)
add urls in the DESCRIPTION file pointing to the java dependency sources

gbisschoff commented:
So I added some more information on the java dependencies and their authors and updated the license. Both java dependencies use the Apache 2.0 license. 

- add more detail to DESCRIPTION file on java dependency authors  …
- change license from GPL-3 to Apache 2.0 (LICENSE file)
- add urls in the DESCRIPTION file pointing to the java dependency sources

jwijffels commented:
@gbisschoff Thanks for the update but this is clearly not enough for CRAN, you need to 

- You need to extract all copyright information of the java files of these source and add them to the new repository (e.g. I just browsed 1 file namely https://github.com/saurfang/spark-sas7bdat/blob/master/src/main/scala/com/github/saurfang/sas/mapreduce/SasInputFormat.scala and it already indicates some copyright which is not included in your pull request) - you need to do this for all files which were combined in the jar
- An example is here https://github.com/bnosac/crfsuite where this R package ships other people's C++ code with a license compatible with the license of the package

Commit:
add COPYRIGHTS

gbisschoff commented:
Okay I see. I added the copyrights into a copyrights file ?

jwijffels commented:
I think you can add them in a LICENSE.note file
Did you look to all .java files of both spark-sas7bdat as well as parso to see if there are any copyright/license statments that should be included?

Note also according to the cran policies at https://cran.r-project.org/web/packages/policies.html that you need to either include the .java files or indicate where you retrieved (which repo+which commit) them in the java directory and how you created the jar file from the .java files

> Source packages under an Open Source license must provide source or something which can easily be converted back to source (e.g., .rda files) for all components of the package (including for example PDF documentation, configure files produced by autoconf). For Java .class and .jar files, the sources should be in a top-level java directory in the source package (or that directory should explain how they can be obtained).
> 
> Such packages are not permitted to require (e.g., by specifying in ‘Depends’, ‘Imports’ or ‘LinkingTo’ fields) directly or indirectly a package or external software which restricts users or usage.
> 
> The package’s license must give the right for CRAN to distribute the package in perpetuity. Any change to a package’s license must be highlighted when an update is submitted (for there have been instances of an undocumented license change removing even the right of CRAN to distribute the package).
